lib/inspec/metadata.rb

```diff
@@ -52,57 +53,25 @@ def supports(sth, version = nil)
       # already.
     end
 
-    def is_supported?(os, entry)
-      name = entry[:'os-name'] || entry[:os]
-      family = entry[:'os-family']
-      release = entry[:release]
-
-      # return true if the backend matches the supported OS's
-      # fields act as masks, i.e. any value configured for os-name, os-family,
-      # or release must be met by the backend; any field that is nil acts as
-      # a glob expression i.e. is true
-
-      # os name is both saved in :family and :name, so check both
-      name_ok = name.nil? ||
-                os[:name] == name || os[:family] == name
-
-      family_check = family.to_s + '?'
-      family_ok = family.nil? || os[:family] == family ||
-                  (
-                    os.respond_to?(family_check) &&
-                    # this call will return true if the family matches
-                    os.method(family_check).call
-                  )
-
-      # ensure we do have a string if we have a non-nil value eg. 16.06
-      release_ok = release.nil? || os[:release] == release
-
-      # we want to make sure that all matchers are true
-      name_ok && family_ok && release_ok
-    end
-
     def inspec_requirement
-      inspec = params[:supports].find { |x| !x[:inspec].nil? } || {}
-      Gem::Requirement.create(inspec[:inspec])
+      inspec_in_supports = params[:supports].find { |x| !x[:inspec].nil? }
+      if inspec_in_supports
+        Inspec::Log.warn '[DEPRECATED] The use of inspec.yml `supports:inspec` is deprecated and will be removed in InSpec 2.0. Please use `inspec_version` instead.'
+        Gem::Requirement.create(inspec_in_supports[:inspec])
+      else
+        # using Gem::Requirement here to allow nil values which
+        # translate to [">= 0"]
+        Gem::Requirement.create(params[:inspec_version])
+      end
     end
 
     def supports_runtime?
       running = Gem::Version.new(Inspec::VERSION)
       inspec_requirement.satisfied_by?(running)
     end
 
-    def supports_transport?(backend)
-      # with no supports specified, always return true, as there are no
-      # constraints on the supported backend; it is equivalent to putting
-      # all fields into accept-all mode
-      return true if params[:supports].empty?
-
-      found = params[:supports].find do |entry|
-        is_supported?(backend.os, entry)
-      end
-
-      # finally, if we found a supported entry, we are good to go
-      !found.nil?
+    def supports_platform?(backend)
+      backend.platform.supported?(params[:supports])
```

Comment:
I am a little bit confused by that. We have a backend (aka train connection) that exposes platform and we have the platform resource. It is really difficult to differentiate this now.

Reply:
Naming things is hard, and we've abused some of that terminology because of the origin of some of the Train code (coming from Test Kitchen, where some of the words mean different things than how we're using them).

I honestly can't think of a different way of differentiating them, and I think it's no less confusing than us having the `os` resource and the `os` exposed by Train. These are implementation details for authors of InSpec than they are for the consumers of it.

@chris-rock do you have suggestions on ways to clarify it?

Reply:
I think it's ok for now. Yeah, it is confusing and we also had it with `os` unfortunately. Unless we find something better of course 😁